include/nil/crypto3/zk/commitments/batched_commitment.hpp

```diff
@@ -104,7 +103,23 @@ namespace nil {
                         return math::lagrange_interpolation(U_interpolation_points);
                     }
 
-                    std::vector<std::vector<typename field_type::value_type>> get_unique_points_list() const{
+                    // We call them singles in recursive verifier
+                    std::vector<typename field_type::value_type> get_unique_points(){
+                        std::vector<typename field_type::value_type> result;
+
+                        for( auto const &[k, point_batch]:_points ){
+                            for( auto const &point_set: point_batch ){
+                                for( auto const &point:point_set ){
+                                    if( std::find(result.begin(), result.end(), point) == result.end() )
```

Comment:
It's very slow to search over a vector. Maybe create a  "std::unordered_set<typename field_type::value_type> result", inserting into it will remove the duplicates. Then convert unordered_set to vector?

If you want to keep the order of elements (you probably don't), you can keep both a vector and an unordered_set, and search inside the set.